tests/testsuite/death.rs

```diff
@@ -87,6 +87,93 @@ fn ctrl_c_kills_everyone() {
     );
 }
 
+#[cargo_test]
+fn kill_never_corrupts_cargo_toml() {
+    use std::str;
+
+    let with_dependency = r#"
+    [package]
+    name = "foo"
+    version = "0.0.1"
+    authors = []
+    build = "build.rs"
+
+    [dependencies]
+    bar = "0.0.1"
+    "#;
+    let without_dependency = r#"
+    [package]
+    name = "foo"
+    version = "0.0.1"
+    authors = []
+    build = "build.rs"
+    "#;
+
+    // This test depends on killing the cargo process at the right time to cause a failed write.
+    // Note that we're iterating and using the index as time in ms to sleep before killing the cargo process.
+    // Max test time ~10s testing timings from 1-100ms between start and kill.
+    // If it is working correctly, we never fail, but can't hang out here all day...
+    for i in 1..100 {
```

Comment:
I'm a bit concerned about having a test that runs for 10s

Reply:
I also do not like.  It's a timing issue, and I'm not sure how else to get the timing right.  Open to suggestions to tighten the loop.  

Is there a regression test suite, or something that run less frequently, but can handle some longer tests?

Reply:
I'm not aware of any stress test suite we can run this with.  I think I'd chalk it up with other ux issues and say "we've tested it as part of the PR".  The pattern is well understood so I don't think we're too concerned on that front.

Reply:
Do these make sense to keep at all?  I reduced to 3 attempts each, for less than half a second total. If not, I'll trim off entirely next.

Reply:
Unsure but the current sleep times might be sufficient.  We'll need to see how we're doing on test time at some point and work to improve it.  If we had proper fixture support, each sleep time could be run in parallel...